shared/js/content-scripts/content-scope-messaging.js

```diff
@@ -10,6 +10,23 @@ function getSecret () {
 async function init () {
     const secret = await getSecret()
 
+    window.addEventListener('sendMessage', event => {
```

Comment:
Is this message coming from our main_world script? If so, couldn't the website spoof such a message too? Perhaps we could use the secret in an event name to mitigate?

Reply:
No the communication from the main world content scripts hasn't changed, it just required some additional proxying internally for MV3

Reply:
OK, so long as you're confident that websites can't just throw a "sendMessage" event on window to trick this logic then it sounds good to me.

Reply:
In fact, could you test that? I am a little concerned this could be a security vulnerability, especially considering even the message type is taken from the event - if it's spoofed that would be bad.

Reply:
OK, I have tested it and it seems to be a security vulnerability. Try going to https://testpages.kzar.co.uk/insecure-extension-messaging/ and the website tell the extension to open its options page + feedback page.

Reply:
It turns out that already exists in the released extension, resolving this therefore.